### PR TITLE
feat: Add "internal page" action

### PR DESCRIPTION
**Pull request by p3rcypj**
### :pushpin: References

-   **Issue:** Closes https://app.clickup.com/t/8693u555f

### :memo: Implementation

Tasks:

- [x] Added flag "Execute on init" to landing pages
- [x] Added action type "Landing page" to be able to go to another landing page

Validations on update and import:
- Action with type "Landing page" cannot be created if the landing page id doesn't exist
- Action with type "Landing page" cannot be added to landing page that references itself on action

### :video_camera: Screenshots/Screen capture

image
image
image

### CODE REVIEW

File: src/data/entities/Config.ts
Comment (p3rcypj) [inline src/data/entities/Config.ts:1]: Clean Arch

File: src/data/entities/Instance.ts
Comment (p3rcypj) [inline src/data/entities/Instance.ts:1]: Clean Arch

File: src/data/entities/JSONAction.ts
Comment (p3rcypj) [inline src/data/entities/JSONAction.ts:1]: Feature

File: src/data/entities/PersistedLandingNode.ts
Comment (p3rcypj) [inline src/data/entities/PersistedLandingNode.ts:1]: Clean Arch

File: src/data/entities/PersistedSettings.ts
Comment (p3rcypj) [inline src/data/entities/PersistedSettings.ts:1]: Clean Arch

File: src/data/repositories/ActionDefaultRepository.ts [hunk 1]
```diff
@@ -179,13 +150,14 @@ export class ActionDefaultRepository implements ActionRepository {
             dhisVersionRange: model.dhisVersionRange,
             dhisAppKey: model.dhisAppKey,
             dhisLaunchUrl: model.dhisLaunchUrl,
+            launchPageId: model.launchPageId,
```
Comment (p3rcypj) [inline src/data/repositories/ActionDefaultRepository.ts:153]: Feature

File: src/data/repositories/ActionDefaultRepository.ts
Comment (p3rcypj) [inline src/data/repositories/ActionDefaultRepository.ts:1]: Clean Arch but one line of Feature L153

File: src/data/repositories/Dhis2ConfigRepository.ts
Comment (p3rcypj) [inline src/data/repositories/Dhis2ConfigRepository.ts:1]: Clean Arch

File: src/data/repositories/InstanceDhisRepository.ts
Comment (p3rcypj) [inline src/data/repositories/InstanceDhisRepository.ts:1]: Clean Arch

File: src/data/repositories/LandingNodeDefaultRepository.ts [hunk 2]
```diff
@@ -82,25 +44,49 @@ export class LandingNodeDefaultRepository implements LandingNodeRepository {
...
+            title: {
+                key: "root-title",
+                referenceValue: "Welcome to Home Page App",
+                translations: {},
+            },
+            content: undefined,
+            actions: [],
+            backgroundColor: "#276696",
+            secondary: false,
+            executeOnInit: true,
```
Comment (p3rcypj) [inline src/data/repositories/LandingNodeDefaultRepository.ts:70]: Feature

File: src/domain/entities/LandingNode.ts
Comment (p3rcypj) [inline src/domain/entities/LandingNode.ts:1]: Feature

File: src/domain/repositories/ActionRepository.ts
Comment (p3rcypj) [inline src/domain/repositories/ActionRepository.ts:1]: Clean Arch

File: src/data/repositories/LandingNodeDefaultRepository.ts
Comment (p3rcypj) [inline src/data/repositories/LandingNodeDefaultRepository.ts:1]: Clean Arch but one line L70 Feature

File: src/data/utils/d2-api.ts
Comment (p3rcypj) [inline src/data/utils/d2-api.ts:1]: Clean Arch

File: src/domain/entities/Action.ts
Comment (p3rcypj) [inline src/domain/entities/Action.ts:1]: Feature

File: src/domain/repositories/ConfigRepository.ts
Comment (p3rcypj) [inline src/domain/repositories/ConfigRepository.ts:1]: Clean Arch

File: src/domain/repositories/InstanceRepository.ts
Comment (p3rcypj) [inline src/domain/repositories/InstanceRepository.ts:1]: Clean Arch

File: src/domain/repositories/LandingNodeRepository.ts
Comment (p3rcypj) [inline src/domain/repositories/LandingNodeRepository.ts:1]: Clean Arch

File: src/domain/usecases/CreateLandingNodeUseCase.ts
Comment (p3rcypj) [inline src/domain/usecases/CreateLandingNodeUseCase.ts:1]: Clean Arch

File: src/domain/usecases/DeleteLandingNodesUseCase.ts
Comment (p3rcypj) [inline src/domain/usecases/DeleteLandingNodesUseCase.ts:1]: Clean Arch

File: src/webapp/pages/home/HomePage.tsx
Comment (p3rcypj) [inline src/webapp/pages/home/HomePage.tsx:1]: Feature

File: src/webapp/pages/App.tsx
Comment (p3rcypj) [inline src/webapp/pages/App.tsx:1]: Clean Arch

File: src/webapp/contexts/app-context.tsx
Comment (p3rcypj) [inline src/webapp/contexts/app-context.tsx:1]: Clean Arch

File: src/webapp/components/landing-page-list-table/LandingPageListTable.tsx
Comment (p3rcypj) [inline src/webapp/components/landing-page-list-table/LandingPageListTable.tsx:1]: Feature

File: src/webapp/components/landing-page-edit-dialog/LandingPageEditDialog.tsx
Comment (p3rcypj) [inline src/webapp/components/landing-page-edit-dialog/LandingPageEditDialog.tsx:1]: Feature

File: src/webapp/components/item-sub-section/ItemSubSection.tsx
Comment (p3rcypj) [inline src/webapp/components/item-sub-section/ItemSubSection.tsx:1]: Feature

File: src/webapp/components/item-section/ItemSection.tsx
Comment (p3rcypj) [inline src/webapp/components/item-section/ItemSection.tsx:1]: Feature

File: src/webapp/components/item-root/ItemRoot.tsx
Comment (p3rcypj) [inline src/webapp/components/item-root/ItemRoot.tsx:1]: Feature

File: src/webapp/components/item-category/ItemCategory.tsx
Comment (p3rcypj) [inline src/webapp/components/item-category/ItemCategory.tsx:1]: Feature

File: src/webapp/components/additional-components/AdditionalComponents.tsx
Comment (p3rcypj) [inline src/webapp/components/additional-components/AdditionalComponents.tsx:1]: Feature

File: src/webapp/components/action-list-table/ActionListTable.tsx
Comment (p3rcypj) [inline src/webapp/components/action-list-table/ActionListTable.tsx:1]: Feature

File: src/webapp/components/action-creation-wizard/steps/index.ts
Comment (p3rcypj) [inline src/webapp/components/action-creation-wizard/steps/index.ts:1]: Feature

File: src/webapp/components/action-creation-wizard/steps/GeneralInfoStep.tsx
Comment (p3rcypj) [inline src/webapp/components/action-creation-wizard/steps/GeneralInfoStep.tsx:1]: Feature

File: src/webapp/components/action-creation-wizard/ActionCreationWizard.tsx
Comment (p3rcypj) [inline src/webapp/components/action-creation-wizard/ActionCreationWizard.tsx:1]: Feature

File: src/webapp/CompositionRoot.ts
Comment (p3rcypj) [inline src/webapp/CompositionRoot.ts:1]: Clean Arch

File: src/domain/usecases/common.ts
Comment (p3rcypj) [inline src/domain/usecases/common.ts:1]: Clean Arch

File: src/domain/usecases/UpdateLandingNodeUseCase.ts
Comment (p3rcypj) [inline src/domain/usecases/UpdateLandingNodeUseCase.ts:1]: Clean Arch

File: src/domain/usecases/UpdateActionUseCase.ts
Comment (p3rcypj) [inline src/domain/usecases/UpdateActionUseCase.ts:1]: Feature

File: src/domain/usecases/SwapLandingChildOrderUseCase.ts
Comment (p3rcypj) [inline src/domain/usecases/SwapLandingChildOrderUseCase.ts:1]: Clean Arch

File: src/domain/usecases/ListLandingChildrenUseCase.ts
Comment (p3rcypj) [inline src/domain/usecases/ListLandingChildrenUseCase.ts:1]: Clean Arch

File: src/domain/usecases/ListActionsUseCase.ts
Comment (p3rcypj) [inline src/domain/usecases/ListActionsUseCase.ts:1]: Clean Arch

File: src/domain/usecases/ImportLandingNodesUseCase.ts
Comment (p3rcypj) [inline src/domain/usecases/ImportLandingNodesUseCase.ts:1]: Clean Arch

File: src/domain/usecases/ImportLandingNodesTranslationsUseCase.ts
Comment (p3rcypj) [inline src/domain/usecases/ImportLandingNodesTranslationsUseCase.ts:1]: Clean Arch

File: src/domain/usecases/ImportActionsUseCase.ts
Comment (p3rcypj) [inline src/domain/usecases/ImportActionsUseCase.ts:1]: Both Clean Arch and Feature

File: src/domain/usecases/GetConfigUseCase.ts
Comment (p3rcypj) [inline src/domain/usecases/GetConfigUseCase.ts:1]: Clean Arch

File: src/domain/usecases/ExportLandingNodesUseCase.ts
Comment (p3rcypj) [inline src/domain/usecases/ExportLandingNodesUseCase.ts:1]: Clean Arch

File: src/domain/usecases/ExportLandingNodesTranslationsUseCase.ts
Comment (p3rcypj) [inline src/domain/usecases/ExportLandingNodesTranslationsUseCase.ts:1]: Clean Arch

File: src/domain/usecases/ExportActionsUseCase.ts
Comment (p3rcypj) [inline src/domain/usecases/ExportActionsUseCase.ts:1]: Clean Arch

**Review by xurxodev — CHANGES_REQUESTED** [review]
Thanks @p3rcypj 

Maybe it's my little knowledge about this project because my last time was a long time ago, but I tested quickly the functionality previous to review it from technical point of view, and it doesn't work.

My test:
url: https://dev.eyeseetea.com/play

- I edit the first action to landing type and select the unique landing page (main) to launch.
- When I click in the action from the home appear a snackbar with the text 'Page not found'

**Comment by p3rcypj** [comment]
@xurxodev Kinda strange, it should say at least as error "You are already in this page". The only thing that comes to my mind is possible the main landing page changing IDs as is the default¿¿ I tried also in [dev.eyeseetea.com/play](dev.eyeseetea.com/play) and yes, it throws error, but the mentioned before that is intended.

The general steps to reproduce the usecase for this feature should be something like: Create new landing with execute on init disabled, put some content or actions on that landing, create a new one action of landing type and select the previous landing, and add that action to the landing that it executes on init.

**Review by xurxodev — APPROVED** [review]
Thanks @p3rcypj 

I've tested the use case that you indicate and works. However, the use case that I indicate fail appearing the page not found message.

Anyway, I leave the functional review to @adrianq 

I've reviewed the PR from an architecture point of view, and I've found some issues about architecture to review. I'm going to approve the PR because it is big, and it has mixed new functionality and architecture refactors. 

Furthermore, I've written the issues to review in the pair programming coding dojo in the task 
https://app.clickup.com/t/86943v5z0

**Comment by adrianq** [comment]
@xurxodev I reviewed the issue this morning with @p3rcypj He has clearly identified where the problem is coming from. It is true this is not the most typical use case but we should be able to handle this problem in a more elegant way. For now, @p3rcypj can focus on the reactors he has discussed with you and create a different PR to fix this issue afterwards

**Pull request merged**